core/EE_Registry.core.php

```diff
@@ -693,48 +755,52 @@ public function create(
         if ($load_only) {
             return true;
         }
-        $addon = $addon
-            ? 'addon'
-            : '';
+        $addon = $addon ? 'addon' : '';
         // $this->_cache_on is toggled during the recursive loading that can occur with dependency injection
         // $cache is controlled by individual calls to separate Registry loader methods like load_class()
         // $load_only is also controlled by individual calls to separate Registry loader methods like load_file()
         if ($this->_cache_on && $cache && ! $load_only) {
             // return object if it's already cached
-            $cached_class = $this->_get_cached_class($class_name, $addon);
+            $cached_class = $this->_get_cached_class($class_name, $addon, $arguments);
```

Comment:
ooo... so I'm guessing this was the source of some of the bugs we saw?

Reply:
unfortunately I think there was more than one source. a bunch of little things that all added up to the error you saw